modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/MuleExtensionUtils.java

```diff
@@ -584,4 +601,70 @@ public static boolean isConnectedStreamingOperation(ComponentModel componentMode
     }
     return java.util.Optional.empty();
   }
+
+  /**
+   * Checks the given {@link ParameterGroupModel}s against the given resolved parameters and validates that the group
+   * exclusiveness is honored.
+   *
+   * @param model                 the model that owns the parameter groups
+   * @param groups                the parameter groups in the model whose exclusiveness must be honored
+   * @param parameters            the resolved parameters
+   * @param aliasedParameterNames the alias for each parameter if it has one
+   *
+   * @throws ConfigurationException if the group exclusiveness is not honored by the resolved parameters.
+   *
+   * @since 4.4.0
+   */
+  public static void checkParameterGroupExclusiveness(Optional<ParameterizedModel> model,
```

Comment:
What is the reason to move this to a util class?

Reply:
previously, it was a public method inside a class but it used no internal fields or state of the class. it checks things over the model so it made sense to me to move it,